lib/jwt/jwk/rsa.rb

```diff
@@ -8,19 +8,21 @@ class RSA
       extend Forwardable
 
       attr_reader :keypair
+      attr_reader :jwk_kid
 
       def_delegators :keypair, :private?, :public_key
 
       BINARY = 2
       KTY    = 'RSA'.freeze
 
-      def initialize(keypair)
+      def initialize(keypair, kid = nil)
         raise ArgumentError, 'keypair must be of type OpenSSL::PKey::RSA' unless keypair.is_a?(OpenSSL::PKey::RSA)
-
+        @jwk_kid = kid
         @keypair = keypair
       end
 
       def kid
+        return jwk_kid if jwk_kid
```

Comment:
Would it make sense to have this method as something like:
```
def kid
  @kid ||= generate_from_public_key
end
```

Then we can just set the instance variable ```@kid``` in the constructor.